code/game/objects/items/devices/laserpointer.dm

```diff
@@ -68,103 +132,133 @@
 			. += span_notice("The diode is missing.")
 		else
 			. += span_notice("A class <b>[diode.rating]</b> laser diode is installed. It is <i>screwed</i> in place.")
+		. += span_notice("A small display reads out that[recharge_locked ? " it is currently recharging to full, and" : ""] there is <b>[energy * 10]%</b> total charge remaining.")
+		if(crystal_lens)
+			. += span_notice("There is a <b>[crystal_lens.name]</b> fit neatly before the focus lens. It can be <i>plucked out</i> with some <i>wirecutters</i>.")
+		else
+			if(diode) //hint at the ability to modify the pointer with a crystal only if we have a diode
+				. += span_notice("<i>You could examine it more thoroughly...</i>")
+
+/obj/item/laser_pointer/examine_more(mob/user)
+	. = ..()
+	if(!crystal_lens && diode)
+		switch(diode.rating)
+			if(1)
+				. += "<i>\The [diode.name] is fit neatly into the casing.</i>"
+			if(2)
+				. += "<i>\The [diode.name] is secured in place, with a little bit of room left between it and the focus lens.</i>"
+			if(3 to 4)
+				. += "<i>\The [diode.name]'s size is much smaller compared to the previous generation lasers, \
+				and the wide margin between it and the focus lens could probably house <b>a crystal</b> of some sort.</i>"
 
 /obj/item/laser_pointer/afterattack(atom/target, mob/living/user, flag, params)
 	. = ..()
 	. |= AFTERATTACK_PROCESSED_ITEM
 	laser_act(target, user, params)
 
+///Handles shining the clicked atom,
 /obj/item/laser_pointer/proc/laser_act(atom/target, mob/living/user, params)
-	if( !(user in (viewers(7,target))) )
-		return
-	if (!diode)
+	if(!diode)
 		to_chat(user, span_notice("You point [src] at [target], but nothing happens!"))
 		return
-	if (!ISADVANCEDTOOLUSER(user))
+	if(!ISADVANCEDTOOLUSER(user))
 		to_chat(user, span_warning("You don't have the dexterity to do this!"))
 		return
 	if(HAS_TRAIT(user, TRAIT_CHUNKYFINGERS))
 		to_chat(user, span_warning("Your fingers can't press the button!"))
 		return
+
+	if(!(user in (viewers(7, target)))) //check if we can see the target (xray and thermals can give us vision through obstacles)
+		return
+	if(!(user in (view(7, target)))) //check if we are visible from the target's PoV
+		if(!crystal_lens)
+			to_chat(user, span_warning("You can't point with [src] through walls!"))
+			return
+		if(!((user.sight & SEE_OBJS) || (user.sight & SEE_MOBS))) //only let it work if we have xray or thermals. mesons don't count because they are easier to get.
+			to_chat(user, span_notice("You can't quite make out your target and you fail to shine at it."))
+			return
```

Comment:
i tried to make it so you need to have xray or thermals to actually be able to point at stuff behind walls, so meson vision won't be plausible. however, combining meson vision with thermal vision (i.e. thermal mutation + meson goggles) does allow targeting any atom, though you won't see the blip. i kinda feel like it's fine, since that meson-thermals combo is rare to come by and in case of the mutation combo it's only available for a short window of time